### ISSUE TITLE
DATABASE: Enhancements for condition zapper 

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#191]
Condition zapper will fail when deleting a large number of conditions because the sql isn't written correctly. We are not sure exactly what this number is. Limit the number of conditions to be deleted to 1000 (or less, whatever builds valid sql) up front. Either only take the first 1000 or don't run unless there are 1000 or less. Either way, inform the user if they try > 1000.

Add logging at each step so if the tool fails, we know how far it got. Currently it writes the log at the end when the process completes successfully.

**Comment by lbross** [comment]
@sorgtyler I added a method to the dao_data_access today called IndexExists(). Turns out I did need it after all for the version_control for v5.8.7. It's currently in the issue 210 branch but will be merged into master when I do the next build. Seems to work as desired.

**Comment by lbross** [comment]
@timgholland received this error when running condition zapper on CEC. @sorgtyler did some research and found the following:

Tim’s screenshot showed PREPOST_FVS_CUSTOM.accdb.
Online, the error is associated with Large Number vs BigInt data types. BigInt is new in 2016 office. Is there a column like this in one if the tables Tim is deleting from?
There might be an issue with the office runtime vs the office version. I vaguely remember having errors like this until I configured my environment correctly.
Or, do you think it has to do with the familiar resource errors from large delete batches? I did add batching code a while back?

![image](https://user-images.githubusercontent.com/11527966/61909403-e63d5100-aee6-11e9-9d8a-e2aa380e60c7.png)

**Comment by lbross** [comment]
I looked in the two tables in CEC PREPOST_FVS_CUSTOM.accdb. (I assume the condition zapper deletes from actual tables rather than links?) I did not see any BigInt data types in those tables. @sorgtyler I can send you a copy of PREPOST_FVS_CUSTOM.accdb if you want to take a look. I think the most likely possibility of the possibilities you provided is and issue with the office runtime vs the office version. My vote is not to pursue this for the moment since the immediate need is gone. But we have the information in GitHub if someone else reports the problem.
